### PR TITLE
Tournament tscheck

### PR DESCRIPTION
**Pull request by LDSGOhGo**
passes tscheck, and is tested.

### CODE REVIEW

**Comment by BobbyTheCatfish** [comment]
Made a few changes to simplify stuff. I've tested it but I want you to test it as well and give me your thoughts

**Review by BobbyTheCatfish — APPROVED** [review]

**Comment by LDSGOhGo** [comment]
Tested, works great just had to fix the replies for the participant command to be edits of a deferred reply.

File: modules/tournament.js [hunk 1]
```diff
@@ -1,94 +1,118 @@
...
   const displayTourneys = [];
   for (const tournament of tournaments) {
-    let displayDate = (tournament.tournament.start_at ? new Date(tournament.tournament.start_at.substr(0, tournament.tournament.start_at.indexOf("T"))) : "Unscheduled");
-    if (typeof displayDate != "string") displayDate = displayDate.toLocaleDateString("en-us");
-    displayTourneys.push(`${displayDate}: [${tournament.tournament.name}](${tournament.tournament.full_challonge_url})`);
+    const displayDate = (tournament.start_at ? u.time(new Date(tournament.start_at), "D") : "Unscheduled");
+    displayTourneys.push(`${displayDate}: [${tournament.name}](${tournament.full_challonge_url})`);
   }
 
   if (displayTourneys.length == 0) return int.editReply({ content: "Looks like there aren't any tourneys scheduled right now." });
```
Comment (BobbyTheCatfish) [inline modules/tournament.js:43]:
```suggestion
  if (displayTourneys.length == 0) return int.editReply("Looks like there aren't any tourneys scheduled right now.");
```

File: modules/tournament.js [hunk 2]
```diff
@@ -1,94 +1,118 @@
...
       const member = members.at(i);
       try {
-        await member.roles.remove(role.id);
+        await member?.roles.remove(role.id);
         succeeded++;
       } catch (error) {null;}
       i++;
     }
-    return int.reply({ content: `Removed ${succeeded}/${members.size} people from the ${role} role`, ephemeral: true });
+    return int.editReply({ content: `Removed ${succeeded}/${members.size} people from the ${role} role` });
```
Comment (BobbyTheCatfish) [inline modules/tournament.js:92]:
```suggestion
    return int.editReply(`Removed ${succeeded}/${members.size} people from the ${role} role`);
```

File: modules/tournament.js [hunk 3]
```diff
@@ -1,94 +1,118 @@
...
   } else if (remove) {
-    if (user.roles.cache.has(role.id)) {
+    if (user?.roles.cache.has(role.id)) {
       let content = `I removed the ${role} role from ${user}`;
       await user.roles.remove(role.id).catch(() => content = `I couldn't remove the ${role} role from ${user}`);
-      return int.reply({ content, ephemeral: true });
+      return int.editReply({ content });
```
Comment (BobbyTheCatfish) [inline modules/tournament.js:97]:
```suggestion
      return int.editReply(content);
```

File: modules/tournament.js [hunk 4]
```diff
@@ -1,94 +1,118 @@
...
     } else {
-      return int.reply({ content: `${user} doesn't have the ${role} role`, ephemeral: true });
+      return int.editReply({ content: `${user} doesn't have the ${role} role` });
```
Comment (BobbyTheCatfish) [inline modules/tournament.js:99]:
```suggestion
      return int.editReply(`${user} doesn't have the ${role} role`);
```

File: modules/tournament.js [hunk 5]
```diff
@@ -1,94 +1,118 @@
...
-  } else if (!user.roles.cache.has(role.id)) {
+  } else if (!user?.roles.cache.has(role.id)) {
     let content = `I added the ${role} role to ${user}`;
-    await user.roles.add(role.id).catch(() => content = `I couldn't add the ${role} role to ${user}`);
-    return int.reply({ content, ephemeral: true });
+    await user?.roles.add(role.id).catch(() => content = `I couldn't add the ${role} role to ${user}`);
+    return int.editReply({ content });
   } else {
-    return int.reply({ content: `${user} already has the ${role} role`, ephemeral: true });
+    return int.editReply({ content: `${user} already has the ${role} role` });
```
Comment (BobbyTheCatfish) [inline modules/tournament.js:106]:
```suggestion
    return int.editReply(`${user} already has the ${role} role`);
```

**Review by BobbyTheCatfish — CHANGES_REQUESTED** [review]
Final few changes, I can apply them

**Review by BobbyTheCatfish — APPROVED** [review]

**Pull request merged**